core/ctk/ctk-filedialog.c

```diff
@@ -155,14 +155,14 @@ ctk_filedialog_eventhandler(struct ctk_filedialog_state *s,
       }
       return 1; 
     } else if(ev == ctk_signal_keypress) {
-      if((ctk_arch_key_t)data == CH_CURS_UP) {
+      if((unsigned long)data == CH_CURS_UP) {
```

Comment:
Why `unsigned long`? The "original" data type is `size_t`:
```
process_post_synch(window->owner, ctk_signal_keypress, (process_data_t)(size_t)c);
```
Apart from that it isn't optimal to comapre (at least) 32 bits when comapring 8 bits is enough. The "correct" solution is present i.e. in calc.c:
```
if(ev == ctk_signal_keypress) {
  if((char)(size_t)data >= '0' &&
     (char)(size_t)data <= '9') {
```